public/pages/workflow_detail/workflow_inputs/processor_inputs/modals/configure_prompt_modal.tsx

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import { useFormikContext, getIn } from 'formik';
+import { isEmpty } from 'lodash';
+import {
+  EuiCodeEditor,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSmallButton,
+  EuiSpacer,
+  EuiText,
+  EuiPopover,
+  EuiSmallButtonEmpty,
+  EuiPopoverTitle,
+  EuiCodeBlock,
+  EuiCode,
+  EuiBasicTable,
+  EuiAccordion,
+  EuiCopy,
+  EuiButtonIcon,
+  EuiContextMenu,
+} from '@elastic/eui';
+import {
+  IProcessorConfig,
+  ModelInputFormField,
+  ModelInterface,
+  PROMPT_PRESETS,
+  PromptPreset,
+  WorkflowFormValues,
+  customStringify,
+} from '../../../../../../common';
+import {
+  parseModelInputs,
+  parseModelInputsObj,
+} from '../../../../../utils/utils';
+
+interface ConfigurePromptModalProps {
+  config: IProcessorConfig;
+  baseConfigPath: string;
+  modelInterface: ModelInterface | undefined;
+  onClose: () => void;
+}
+
+/**
+ * A modal to configure a prompt template. Can manually configure, include placeholder values
+ * using other model inputs, and/or select from a presets library.
+ */
+export function ConfigurePromptModal(props: ConfigurePromptModalProps) {
+  const { values, setFieldValue, setFieldTouched } = useFormikContext<
+    WorkflowFormValues
+  >();
+
+  // get some current form values
+  const modelConfigPath = `${props.baseConfigPath}.${props.config.id}.model_config`;
+  const modelConfig = getIn(values, modelConfigPath) as string;
+  const modelInputs = parseModelInputs(props.modelInterface);
+
+  // popover states
+  const [schemaPopoverOpen, setSchemaPopoverOpen] = useState<boolean>(false);
+  const [presetsPopoverOpen, setPresetsPopoverOpen] = useState<boolean>(false);
+
+  // prompt str state. manipulated as users manually update, or
+  // from selecting a preset
+  const [promptStr, setPromptStr] = useState<string>('');
+
+  // hook to set the prompt if found in the model config
+  useEffect(() => {
+    const modelConfigString = getIn(
+      values,
+      `${props.baseConfigPath}.${props.config.id}.model_config`
+    ) as string;
+    try {
+      const prompt = JSON.parse(modelConfigString)?.prompt;
+      if (!isEmpty(prompt)) {
+        setPromptStr(prompt);
+      } else {
+        setPromptStr('');
+      }
```

Comment:
@ohltyler, shall we change it to below code?
```
const prompt = JSON.parse(modelConfigString)?.prompt ?? '';
setPromptStr(prompt);
```

Reply:
[isEmpty](https://www.geeksforgeeks.org/lodash-_-isempty-method/) handles undefined, which is what is returned if no `prompt` field is found, so we don't need to add a fallback to convert to empty string.
```
Lodash _.isEmpty() checks if a value is considered empty. This includes objects with no own properties, arrays with no elements, strings of length zero, and other values like null, undefined, and false.
```